lib/util/Components.js

```diff
@@ -442,7 +449,7 @@ function componentRule(rule, context) {
           return null;
         }
         // Return the node if it is a function that is not a class method and is not inside a JSX Element
-        if (isFunction && !isMethod && !isJSXExpressionContainer) {
+        if (isFunction && !isMethod && !isJSXExpressionContainer && utils.isReturningJSX(node)) {
```

Comment:
does `isReturningJSX` return true when something returns null?

Also, in React 16, components can return strings or arrays, so I'm not sure this detection will be reliable moving forward.

"React 16 fragments" will probably be something we have to solve in a separate PR anyways tho, so no need to address that now (i filed #1510)